cmd/kubeapps-apis/proto/kubeappsapis/core/packages/v1alpha1/packages.proto

```diff
@@ -12,124 +12,249 @@ import "protoc-gen-openapiv2/options/annotations.proto";
-  // all plugins may support this back reference (eg. kapp-controller)
-  PackageRepositoryReference repository = 4;
-
-  // Package repository plugin
-  //
-  // The plugin used to interact with this package repository.
-  kubeappsapis.core.plugins.v1alpha1.Plugin plugin = 5;
+  // List of AvailablePackageDetail
+  AvailablePackageDetail available_package_detail = 1;
 }
 
-// GetAvailablePackages
+// GetPackageRepositories
 //
-// Response for GetAvailablePackages
-message GetAvailablePackagesResponse {
+// Response for GetPackageRepositories
+message GetPackageRepositoriesResponse {
   option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_schema) = {
-    example: '{"packages": [{"name": "app1.example.com", "version": "9.10.09", "iconUrl": ""}]}'
+    example: '{"repositories": [{"name": "repo-name.example.com", "namespace": "", "url": "foo.registry.example.com/repo-name/main@sha256:cecd9b51b1f29a773a5228fe04faec121c9fbd2969de55b0c3804269a1d57aa5"}]}'
   };
-  // List of AvailablePackage
-  repeated AvailablePackage packages = 1;
+
+  // Repositories
+  //
+  // List of PackageRepository
+  repeated PackageRepository repositories = 1;
 }
 
-// GetPackageRepositories
+// -- Start definitions used in the responses  -- 
+
+// AvailablePackageSummary
 //
-// Request for GetPackageRepositories
-message GetPackageRepositoriesRequest {
-  // A cluster name can be provided if multiple clusters are configured,
-  // otherwise the current cluster will be assumed.
-  string cluster = 1;
+// An AvailablePackageSummary provides a summary of a package available for installation
+// useful when aggregating many available packages.
+message AvailablePackageSummary {
+  // TODO: add example for API docs
+  // option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_schema) = {
+  //   example: '{}'
+  // };
 
-  // A namespace can be provided if the package repositories for a specific namespace
-  // are requested, when supported by the packaging format.
-  string namespace = 2;
+  // Available package reference
+  //
+  // A reference uniquely identifying the package.
+  AvailablePackageReference available_package_ref = 1;
+  
+  // Latest available package version
+  //
+  // The latest version available for this package. Often expected when viewing
+  // a summary of many available packages.
+  string latest_version = 2;
+  
+  // Available package Icon URL
+  //
+  // A url for an icon.
+  string icon_url = 3;
+  
+  // Available package display name
+  //
+  // A name as displayed to users
+  string display_name = 4;
+  
+  // Available package short description
+  //
+  // A short description of the app provided by the package
+  string short_description = 5;
+}
+
+// AvailablePackageDetail
+//
+// An AvailablePackageDetail provides additional details required when
+// inspecting an individual package.
+message AvailablePackageDetail {
+  // TODO: add example for API docs
```

Comment:
Does the example for API docs make sense when the entity is a read-only returned entity (ie. the user never constructs this message).

Reply:
Well, good point: it is not required or indispensable whatsoever, but, from the API client perspective, I find it really useful to have examples of the returned values. This way is even simpler for clients to mock the returned values when building the API client. 
We are providing the schema, which should be more than enough, but I personally like to provide real examples.

See what you think here, as usual, no strong opinion though.

Reply:
Yep, if the example is also useful as a response, not just an input, great.